pkg/handler/enqueue.go

```diff
@@ -52,25 +53,47 @@ func (e *TypedEnqueueRequestForObject[T]) Create(ctx context.Context, evt event.
 		enqueueLog.Error(nil, "CreateEvent received with no metadata", "event", evt)
 		return
 	}
-	q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
+
+	item := reconcile.Request{NamespacedName: types.NamespacedName{
 		Name:      evt.Object.GetName(),
 		Namespace: evt.Object.GetNamespace(),
-	}})
+	}}
+
+	priorityQueue, isPriorityQueue := q.(priorityqueue.PriorityQueue[reconcile.Request])
+	if !isPriorityQueue {
+		q.Add(item)
+		return
+	}
```

Comment:
Should we also move this into the util func? (and rename the func to addToQueueCreate)

Same for update

Reply:
I can. I would have to move `isObjectUnchanged` as well.  Not sure where to move it if we want that exposed or in some internal directory.

Reply:
Gotcha, I misunderstood.  I can move them as well to do the check within the function. I did rename them but can move those lines inside.

Reply:
Sounds good. Just a lot of code we can deduplicate :)